### PR TITLE
Add an annotation to expose transforms to yaml.

### PR DESCRIPTION
**Pull request by robertwb**
We should add this to all transforms that are simply parameterized.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

**Comment by robertwb** [comment]
R: @Polber

File: sdks/python/apache_beam/yaml/yaml_transform_test.py [hunk 1]
```diff
@@ -229,6 +229,24 @@ def test_name_is_ambiguous(self):
             output: AnotherFilter
             ''')
 
+  def test_annotations(self):
+    t = LinearTransform(5, b=100)
+    annotations = t.annotations()
+    print(annotations)
```
Comment (Polber) [inline sdks/python/apache_beam/yaml/yaml_transform_test.py:235]: Is the `print` meant to still be here?

**Review by Polber — COMMENTED** [review]
Will this replace the annotation assembly performed in `yaml_transform.py`? This would require all transforms used in a YAML pipeline have this annotation, right?

**Comment by robertwb** [comment]
This won't replace the assembly performed in `yaml_transform.py, this merely gives indications of how to use other (posibly not-directly-exposed) Beam transforms directly from yaml.

File: sdks/python/apache_beam/yaml/yaml_transform_test.py (hunk 1 above)
Comment (robertwb) [reply sdks/python/apache_beam/yaml/yaml_transform_test.py:235]: good catch

**Comment by robertwb** [comment]
Any concerns about this?

**Review by Polber — APPROVED** [review]
No concern, just wondering the use-case since the annotations get added in `yaml_transform`, but this makes sense for custom transforms/ those not exposed to yaml as you said. LGTM

**Pull request merged**